src/drivers/intel/dmic/dmic.c

```diff
@@ -248,6 +260,16 @@ static int dmic_set_config(struct dai *dai, struct ipc_config_dai *common_config
 
 	dcache_writeback_region(dmic_prm[di], sizeof(struct sof_ipc_dai_dmic_params));
 	ret = dmic_set_config_computed(dai);
+
+#elif CONFIG_INTEL_DMIC_NHLT
```

Comment:
can we do it without this macro definition ?  In this function parameter struct ipc_config_dai *common_config  there is a flag : is_config_blob, so we can decide which path we can go

Reply:
Yes, think the macros could be preserved to optionally save code size by omitting undesired part, but could add enable both for run-time possibility.